packages/flutter/lib/src/material/app.dart

```diff
@@ -938,52 +936,45 @@ class _MaterialAppState extends State<MaterialApp> {
   Widget _exitWidgetSelectionButtonBuilder(
     BuildContext context, {
     required VoidCallback onPressed,
+    required String semanticLabel,
     required GlobalKey key,
   }) {
-    return FloatingActionButton(
-      key: key,
+    return _MaterialInspectorButton(
       onPressed: onPressed,
-      mini: true,
-      backgroundColor: _widgetSelectionButtonsBackgroundColor(context),
-      foregroundColor: _widgetSelectionButtonsForegroundColor(context),
-      child: const Icon(Icons.close, semanticLabel: 'Exit Select Widget mode.'),
+      semanticLabel: semanticLabel,
+      icon: Icons.close,
+      isDarkTheme: _isDarkTheme(context),
+      buttonKey: key,
     );
   }
 
   Widget _moveExitWidgetSelectionButtonBuilder(
     BuildContext context, {
     required VoidCallback onPressed,
+    required String semanticLabel,
     bool isLeftAligned = true,
   }) {
-    return IconButton(
-      color: _widgetSelectionButtonsBackgroundColor(context),
-      padding: EdgeInsets.zero,
-      iconSize: _moveExitWidgetSelectionIconSize,
+    return _MaterialInspectorButton.iconOnly(
       onPressed: onPressed,
-      constraints: const BoxConstraints(
-        minWidth: _moveExitWidgetSelectionTargetSize,
-        minHeight: _moveExitWidgetSelectionTargetSize,
-      ),
-      icon: Icon(
-        isLeftAligned ? Icons.arrow_right : Icons.arrow_left,
-        semanticLabel:
-            'Move "Exit Select Widget mode" button to the ${isLeftAligned ? 'right' : 'left'}.',
-      ),
+      semanticLabel: semanticLabel,
+      icon: isLeftAligned ? Icons.arrow_right : Icons.arrow_left,
+      isDarkTheme: _isDarkTheme(context),
     );
   }
 
-  Color _widgetSelectionButtonsForegroundColor(BuildContext context) {
-    final ThemeData theme = Theme.of(context);
-    return _isDarkTheme(context)
-        ? theme.colorScheme.onPrimaryContainer
-        : theme.colorScheme.primaryContainer;
-  }
-
-  Color _widgetSelectionButtonsBackgroundColor(BuildContext context) {
-    final ThemeData theme = Theme.of(context);
-    return _isDarkTheme(context)
-        ? theme.colorScheme.primaryContainer
-        : theme.colorScheme.onPrimaryContainer;
+  Widget _tapBehaviorButtonBuilder(
+    BuildContext context, {
+    required VoidCallback onPressed,
+    required String semanticLabel,
+    required bool defaultTapBehaviorEnabled,
+  }) {
+    return _MaterialInspectorButton.toggle(
+      onPressed: onPressed,
+      semanticLabel: semanticLabel,
+      icon: CupertinoIcons.cursor_rays,
```

Comment:
same comment about commenting that this icon matches what is used in DevTools. That might also help explain why we are using a cupertino icon in a material library here